tests/core/math/test_vector4i.h

```diff
@@ -90,6 +90,12 @@ TEST_CASE("[Vector4i] Length methods") {
 	CHECK_MESSAGE(
 			vector2.length() == doctest::Approx(73.4846922835),
 			"Vector4i length should work as expected.");
+	CHECK_MESSAGE(
+			vector1.distance_to(vector2) == doctest::Approx(54.772255750517),
+			"Vector4i distance_to should work as expected.");
+	CHECK_MESSAGE(
+			vector1.distance_squared_to(vector2) == doctest::Approx(3000),
+			"Vector4i distance_squared_to should work as expected.");
```

Comment:
```suggestion
	CHECK_MESSAGE(
			vector1.distance_squared_to(vector2) == 3000,
			"Vector4i distance_squared_to should work as expected.");
	CHECK_MESSAGE(
			vector1.distance_to(vector2) == doctest::Approx(54.772255750517),
			"Vector4i distance_to should work as expected.");
```
Assuming there's no reason to have approximate checking here, keep same order as the others

Reply:
Honestly beats me what's the reason for using `doctest::Approx(3000)`, I was mostly trying to keep implementations identical in parallel to their non-int counterparts:

https://github.com/godotengine/godot/blob/b1371806ad3907c009458ea939bd4b810f9deb21/tests/core/math/test_vector4.h#L102-L107

Approximating here isn't the case for `Vector2` and `Vector3`.

Reply:
But approximation doesn't make sense for integers, so it creates confusion

Reply:
That test seems to originate from #64108, no rationale for using an approximation was given, so I guess it's an error on the author's part coming from copying the `distance_to`'s test's implementation.

That being said I agree that we should go with your suggested changes, but if we do we should also update `godot/tests/core/math/test_vector4.h` accordingly.

Reply:
> but if we do we should also update `godot/tests/core/math/test_vector4.h` accordingly.

Why? It isn't using an integer type, but a floating point type, the `distance_squared_to` returns `int64_t` for `Vector4i`, but `real_t` for `Vector4, so there's no reason to change the `Vector4` case.

Reply:
Okay, but if that's our reasoning then both `Vector2` and `Vector3`'s tests need an update:

https://github.com/godotengine/godot/blob/b1371806ad3907c009458ea939bd4b810f9deb21/tests/core/math/test_vector2.h#L147-L149

https://github.com/godotengine/godot/blob/b1371806ad3907c009458ea939bd4b810f9deb21/tests/core/math/test_vector3.h#L164-L166

Here, we don't really need `doctest::Approx()` whilst testing `distance_squared_to()`, given that these tests would pass without it, but I agree with you that it's more appropriate to use an approximation in the context of `Vector2/3/4`.

Reply:
What I'm saying is that those checks don't need to be changed now, it doesn't matter that they use or don't use the approximate, but that we 100% do not need it for integer types